src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -236,5 +268,110 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
+        /// <returns>comma-separated preferences that can be passed down to an individual request</returns>
+        private static string GetPreferencesToInheritFromBatch(string batchPreferences)
+        {
+            IEnumerable<string> preferencesToInherit = batchPreferences.SplitPreferences()
+                .Where(value => 
+                !nonInheritablePreferences.Any(
+                    prefToIgnore => value.Trim().ToLowerInvariant().StartsWith(prefToIgnore)))
+                .Select(value => value.Trim());
+            return string.Join(", ", preferencesToInherit);
+        }
+
+        /// <summary>
+        /// Merges the preferences from the batch request and an individual request inside the batch into one value.
+        /// If a given preference is defined in both the batch and individual request, the one from the individual
+        /// request is retained and the one from the batch is discarded.
+        /// </summary>
+        /// <param name="individualPreferences">The value of the Prefer header from the individual request inside the batch</param>
+        /// <param name="batchPreferences">The value of the Prefer header from the overall batch request</param>
+        /// <returns>Value containing the combined preferences</returns>
+        private static string MergeIndividualAndBatchPreferences(string individualPreferences, string batchPreferences)
+        {
+            if (string.IsNullOrEmpty(individualPreferences))
+            {
+                return batchPreferences;
+            }
+
+            if (string.IsNullOrEmpty(batchPreferences))
+            {
+                return individualPreferences;
+            }
+            // get the name of each preference to avoid adding duplicates from batch
+            IEnumerable<string> individualList = individualPreferences.SplitPreferences().Select(pref => pref.Trim());
+            HashSet<string> individualPreferenceNames = new HashSet<string>(individualList.Select(pref => pref.Split('=').FirstOrDefault()));
+
+            
+            IEnumerable<string> filteredBatchList = batchPreferences.SplitPreferences().Select(pref => pref.Trim())
+                // do not add duplicate preferences from batch
+                .Where(pref => !individualPreferenceNames.Contains(pref.Split('=').FirstOrDefault()));
+            string filteredBatchPreferences = string.Join(", ", filteredBatchList);
+
+            return string.Join(", ", individualPreferences, filteredBatchPreferences);
+        }
+
+        /// <summary>
+        /// Splits the value of a Prefer header into separate preferences
+        /// e.g. a value like 'a, b=c, foo="bar,baz"' will return an IEnumerable with
+        /// - a
+        /// - b=c
+        /// - foo="bar,baz"
+        /// </summary>
+        /// <param name="preferences"></param>
+        /// <returns></returns>
+        private static IEnumerable<string> SplitPreferences(this string preferences)
```

Comment:
I wouldn't do this as an extension method (especially since it would then apply to all strings).  Instead, just make it a static method that takes a string parameter:
private static IEnumerable<string> SplitPreferences(string preferences) {...}